### PR TITLE
Enabled PhysX system component in asset builders since it's required …

### PR DESCRIPTION
**Pull request by SergeyAMZN**
…for cooking collision meshes

https://github.com/o3de/o3de/issues/2651

Signed-off-by: pereslav <pereslav@amazon.com>

### CODE REVIEW

**Comment by hultonha** [comment]
Ah I see the logic has flipped, we now have `AssetDatabaseService` depending on the `PhysX::SystemComponent` as opposed to the other way around, this looks good to me (if there is a way to validate this with a test of some sort that would be super cool, but could be a follow-up task after this critical fix, would be cool if possible though).

**Review by hultonha — APPROVED** [review]
LGTM :+1:

**Comment by lumberyard-employee-dm** [comment]
> 
> 
> Ah I see the logic has flipped, we now have `AssetDatabaseService` depending on the `PhysX::SystemComponent` as opposed to the other way around, this looks good to me (if there is a way to validate this with a test of some sort that would be super cool, but could be a follow-up task after this critical fix, would be cool if possible though).

Actually the logic isn't flipped.
The difference between a required service and a dependent service, is that a dependent service is optional.
The `PhysX::SystemComponent` has been changed to have a weak dependency on the `AssetDatabaseService`.

Pretty much all this means, is that if the AssetManagerComponent is available on, then the PhysX::SystemComponent will activate after it.

File: Gems/PhysX/Code/Source/SystemComponent.cpp [hunk 1]
```diff
@@ -122,13 +123,13 @@ namespace PhysX
...
-    void SystemComponent::GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+    void SystemComponent::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
     {
-        required.push_back(AZ_CRC("AssetDatabaseService", 0x3abf5601));
     }
 
-    void SystemComponent::GetDependentServices([[maybe_unused]]AZ::ComponentDescriptor::DependencyArrayType& dependent)
+    void SystemComponent::GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent)
     {
+        dependent.push_back(AZ_CRC("AssetDatabaseService", 0x3abf5601));
```
Comment (lumberyard-employee-dm) [inline Gems/PhysX/Code/Source/SystemComponent.cpp:132]: It looks like there is also a dependency on the AssetCatalogService as well in this component

Comment (SergeyAMZN) [reply Gems/PhysX/Code/Source/SystemComponent.cpp]: Added now, thanks!

**Comment by hultonha** [comment]
> 
> 
> > Ah I see the logic has flipped, we now have `AssetDatabaseService` depending on the `PhysX::SystemComponent` as opposed to the other way around, this looks good to me (if there is a way to validate this with a test of some sort that would be super cool, but could be a follow-up task after this critical fix, would be cool if possible though).
> 
> Actually the logic isn't flipped.
> The difference between a required service and a dependent service, is that a dependent service is optional.
> The `PhysX::SystemComponent` has been changed to have a weak dependency on the `AssetDatabaseService`.
> 
> Pretty much all this means, is that if the AssetManagerComponent is available on, then the PhysX::SystemComponent will activate after it.

Ah I see got you

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**